tests/frontier/precompiles/test_precompiles.py

```diff
@@ -0,0 +1,107 @@
+import pytest
+
+from ethereum_test_tools import (
+    Account,
+    Address,
+    Alloc,
+    Environment,
+    StateTestFiller,
+    Transaction,
+)
+from ethereum_test_tools.vm.opcode import Opcodes as Op
+
+
+def precompile_addresses(fork):
+    """
+    Return the precompile addresses for the given fork.
+
+    Args:
+        fork (str): The fork name.
+
+    Returns:
+        list[tuple[str, bool]]: The precompile addresses and whether they exist.
+
+    """
+    return [
+        ("01", True),
+        ("02", True),
+        ("03", True),
+        ("04", True),
+        ("05", True),
+        ("06", True),
+        ("07", True),
+        ("08", True),
+        ("09", True),
+        ("0A", True) if fork == "cancun" else ("0A", False),
+        ("0B", False),
+    ]
+
+
+@pytest.mark.parametrize_by_fork("address,precompile_exists", precompile_addresses)
+def test_precompiles(
+    state_test: StateTestFiller, pre: Alloc, address: str, precompile_exists: bool
+):
+    """Test the MODEXP precompile."""
+    env = Environment()
+
+    address_10000 = 0x10000
+
+    input_buf = 0x1000
+    output_buf = 0x2000
+
+    args_offset = 0x1000
+    args_size = 0x20
+    output_offset = 0x2000
+    output_size = 0x20
+
+    gas_test = 0x00
+    gas_10000 = 0x20
+
+    sender = pre.fund_eoa()
+    account = pre.deploy_contract(
+        Op.MSTORE(args_size, 0x20)
+        + Op.MSTORE(input_buf, 0xFF)
+        + Op.MSTORE(output_buf, 0xFF)
+        + Op.MSTORE8(input_buf, 0xFF)
+        + Op.MSTORE8(output_buf, 0xFF)
+        + Op.MSTORE(gas_test, Op.GAS())
+        # Setup stack to CALL into precompile with the CALLDATA and CALL into it (+ pop value)
+        + Op.CALL(
+            address=Op.CALLDATALOAD(0),
```

Comment:
Trying to pass in the precompile address in the `data` of the transaction for this call. It doesn't seem like I can pass the correct value from the bytes value passed, even if I try to convert the value from the `CALLDATALOAD` to an integer. Is there a recommendation for how to approach this?

Reply:
here the data must be padded as EVM word of 32 bytes. so not enough just fromhex(1 byte) you need to padd it with zeros to the left. like Hash32(data, left_padding=true)